index.js

```diff
@@ -38,6 +42,95 @@ function filterContentSources(content, stream) {
     }
 }
 
+function filterMsidFromRecvonlySources(description) {
+    description.contents.forEach(function(content) {
+        content.application.sources.forEach(function(source) {
+            source.parameters = source.parameters.filter(function(param) {
+                if (param.key === 'msid' && (description.senders ==='initiator' || description.senders ==='none')) {
+                    return false;
+                }
+                return true;
+            });
+        });
+    });
+
+    return description;
+}
+
+function getContent(content) {
+    return content.application.applicationType === 'rtp'
+        && content.application.sources
+        && content.application.sources.length;
+}
+
+function generateDifferenceOfSources(oldLocalDescription, newLocalDescription) {
+    const oldContents = oldLocalDescription.contents.filter(getContent);
+    const newContents = newLocalDescription.contents.filter(getContent);
+
+    const sourceMap = {};
+    const sourcesRemoved = [];
+    const sourcesModified = [];
+
+    //creating map of sources and directions
+    for (var i = 0; i < newContents.length; i++) {
+        for(var j = 0; j < newContents[i].application.sources.length; j++) {
+            sourceMap[newContents[i].application.sources[j].ssrc] = {
+                source: newContents[i].application.sources[j],
+                direction: newContents[i].senders,
+            };
+        }
+    };
+
+
+    for (var i = 0; i < oldContents.length; i++) {
+        for(var j = 0; j < oldContents[i].application.sources.length; j++) {
+            if (!sourceMap[oldContents[i].application.sources[j].ssrc]) {
+                // this IS a removed ssource
+                sourcesRemoved.push(oldContents[i].application.sources[j].ssrc);
+            }
+            else {
+                // this is a possible modified source
+                const oldContentHasMsid = sourceHasMsid(oldContents[i].application.sources[j]);
+                const oldContentSourceDirection = oldContents[i].senders;
+                const newContentHasMsid = sourceHasMsid(sourceMap[oldContents[i].application.sources[j].ssrc].source);
+                const newContentSourceDirection = sourceMap[oldContents[i].application.sources[j].ssrc].direction;
+
+                if ((newContentSourceDirection !== oldContentSourceDirection) || (oldContentHasMsid !== newContentHasMsid)) {
+                    delete oldContents[i].transport;
+                    delete oldContents[i].application.payloads;
+                    sourcesModified.push(oldContents[i].application.sources[j].ssrc);
+                }
+            }
+            delete sourceMap[oldContents[i].application.sources[j].ssrc];
+        }
+    }
+    const sourcesAdded = Object.keys(sourceMap);
+
+    return { sourcesRemoved: sourcesRemoved, sourcesAdded: sourcesAdded, sourcesModified: sourcesModified};
+}
+
+
+
+function getProperSSRCS(contents, ssrcList) {
```

Comment:
This name is confusing and doesn't really indicate what it does

Reply:
changed to getCleanedContentBlockRelatedToSSRCS